src/base/test/test_string.cc

```diff
@@ -159,6 +159,9 @@ TESTCASE(string_formatting) {
 	check_equal("A123      X", format("A%-9.0fX", 123.456));
 	check_equal("A+123.5   X", format("A%+-9.1fX", 123.456));
 	check_equal("A+00123.46X", format("A%0+9.2fX", 123.456));
+	check_equal("A1X", format("A%1%X", 1.0));
+	check_equal("A1X", format("A%1%X", 1.0000002));
```

Comment:
I'm not sure this is desirable. A floating-point argument should always be displayed as a floating-point value with at least one decimal, so IMHO `"A1.0X"` would be the desired outcome for dynamic-precision placeholders.


---

The documentation in tree.h should be updated to mention the behaviour when not specifying an explicit precision for a float node.

Reply:
Unfortunately this is required to fix the encyclopedia entries in the linked issue. Would you allow them to be "5.0%"?

Reply:
Sorry, do you mean OK to leave it as it is, or OK to make it "5.0%"?

Reply:
OK for "5.0%" :smile: